lib/xebow.ex

```diff
@@ -100,33 +121,66 @@ defmodule Xebow do
   # Server Implementations:
 
   @impl GenServer
-  def init(types) do
-    count_of_active_animations = length(types)
-
-    active_animations =
-      types
-      |> Stream.map(&initialize_animation/1)
-      |> Stream.with_index()
-      |> Stream.map(fn {animation, index} -> {index, animation} end)
-      |> Enum.into(%{})
+  def init(_) do
+    active_animation_types =
+      if Settings.active_animations_file_exists?() do
+        case Settings.load_active_animations() do
+          {:ok, active_animation_types} ->
+            active_animation_types
+
+          {:error, reason} ->
+            Logger.warn("Failed to load active animations: #{inspect(reason)}")
+            Animation.types()
+        end
+      else
+        Animation.types()
+      end
 
-    state = %State{
-      current_index: 0,
-      active_animations: active_animations,
-      count_of_active_animations: count_of_active_animations
-    }
+    state = update_state_with_animation_types(%State{}, active_animation_types)
 
-    Engine.set_animation(current_animation(state))
+    case current_animation(state) do
+      nil -> nil
+      animation -> Engine.set_animation(animation)
+    end
 
     {:ok, state}
   end
 
+  @impl GenServer
+  def handle_call(:get_active_animation_types, _from, state) do
+    active_animation_types =
+      state.active_animations
+      |> Map.values()
+      |> Enum.map(& &1.type)
+
+    {:reply, active_animation_types, state}
+  end
+
   @impl GenServer
   def handle_call(:get_animation_config, _from, state) do
-    config = Animation.get_config(current_animation(state))
+    config =
+      case current_animation(state) do
+        nil -> nil
+        animation -> Animation.get_config(animation)
+      end
+
     {:reply, config, state}
   end
 
+  @impl GenServer
+  def handle_cast({:set_active_animation_types, active_animation_types}, state) do
+    state = update_state_with_animation_types(state, active_animation_types)
```

Comment:
:warning: :beetle: (recommended)

This call resets all active animation settings. This means that when you enable or disable an animation, the settings for all of them get reset to default.

So, not necessarily a blocking issue, but definitely a bug that needs to be logged and fixed. If it gets fixed in this PR, that would be really nice, but it's not enough to keep this PR from being merged.

Reply:
:sweat_smile: Good catch. Fixed in cd1accf.

Reply:
Or not quite...

There was also a pre-existing bug in here. Fixed in #114.